ui/src/views/Hooks/ListHooks/index.jsx

```diff
@@ -118,13 +118,9 @@ export default class ListHooks extends Component {
           );
         })
       : [];
-    const handleToggle = (event, nodeIds) => {
-      this.setState({ expanded: nodeIds });
-    };
-
-    const handleSelect = (event, nodeIds) => {
-      this.setState({ selected: nodeIds });
-    };
+    const handleState = stateName => (event, nodeIds) =>
+      this.setState({ [stateName]: nodeIds });
+    const { expanded, selected } = this.state;
```

Comment:
Its not convention to have a function like this in render unless there's some reason it needs to be here. If should be moved into the body of the class.

Reply:
can we have that as a follow up ticket? I would be adding a generic treeview component instead, this one is limited to depth 2 so it might be confusing, alternatively I could just move it as a functional component on top of the same file so we achieve short term rendering performance, what do you think?